dvc/tree/http.py

```diff
@@ -141,7 +141,13 @@ def _head(self, url):
         return response
 
     def exists(self, path_info, use_dvcignore=True):
-        return bool(self._head(path_info.url))
+        res = self._head(path_info.url)
+        if res.status_code == 404:
+            return False
+        if bool(res):
+            return True
+        res.raise_for_status()
```

Comment:
It'd be better if we could wrap this with `HttpError` of dvc's itself.

Reply:
hmm perhaps by using a `try`...`catch` block to do that?

Reply:
or just raising dvc's `HTTPError` directly